src/allmydata/web/root.py

```diff
@@ -133,6 +133,8 @@ def getChild(self, name, req):
         and creates and appropriate handler (depending on the kind of
         capability it was passed).
         """
+        if not name:
+            return self
```

Comment:
If `name` is blank (e.g. user fetched `/uri/` or `/uri`) then we want to return this generic page -- that is, `render_GET` on this object itself. But if there's a trailing slash it seems twisted will ask for the child named `""`. Or I guess: why do you think it's not required?

(I added a clarifying comment).

Reply:
That sort of makes sense although I think it's probably an error prone approach.  There should be one canonical URL for the resource - not multiple aliases.  Aliases are problematic for future URL generation (particularly aliases that differ in whether they end with a `/`).  They create more cases that have to be handled for generated URLs to be correct.

So, for example, if there is a desire for `/uri` to be valid then it should probably be a redirect to `/uri/` and the real resource should live at `/uri/`.

Maybe this is all moot because the code changes here are just supporting existing behavior, though?

Separately from that, `return self` does nothing to stop traversal so while it might make it easier to handle `/uri/?...` and `/uri?...`  it also duplicates the whole `/uri` resource hierarchy beneath `/uri/` (when it really only means to duplicate `/uri` at `/uri/`).  For example, `/uri/////?...` is also valid with this implementation.

Reply:
Okay, I've switched it to a re-direct